frontend/src/modules/Metadata_Forms/components/MetadataFormFields.js

```diff
@@ -0,0 +1,424 @@
+      dispatch({ type: SET_ERROR, error });
+    }
+    setLoading(false);
+  };
+
+  const saveChanges = async (updatedFields) => {
+    setLoading(true);
+    // remove new fields (not yet saved in DB), that were deleted during editing
+    const data = updatedFields.filter((field) => !field.deleted || field.uri);
+    data.forEach((field) => {
+      delete field.__typename;
+    });
+    const response = await client.mutate(
+      batchMetadataFormFieldUpdates(metadataForm.uri, data)
+    );
+    if (
+      !response.errors &&
+      response.data &&
+      response.data.batchMetadataFormFieldUpdates !== null
+    ) {
+      setFields(response.data.batchMetadataFormFieldUpdates);
+      setEditOn(false);
+    } else {
+      const error = response.errors
+        ? response.errors[0].message
+        : 'Update failed';
+      dispatch({ type: SET_ERROR, error });
+    }
+    setLoading(false);
+  };
+
+  const handleInputKeyup = (event) => {
+    if (event.code === 'Enter') {
+      fetchItems().catch((e) =>
+        dispatch({ type: SET_ERROR, error: e.message })
+      );
+    }
+  };
+
+  return (
+    <Box>
+      <Card>
+        <Box
+          sx={{
+            alignItems: 'center',
+            display: 'flex',
+            flexWrap: 'wrap',
+            m: -1,
+            p: 2
+          }}
+        >
+          <TextField
+            disabled="true"
+            fullWidth
+            InputProps={{
+              startAdornment: (
+                <InputAdornment position="start">
+                  <SearchIcon fontSize="small" />
+                </InputAdornment>
+              )
+            }}
+            onChange={handleInputChange}
+            onKeyUp={handleInputKeyup}
+            placeholder="Search (temporary deisabled)"
+            value={inputValue}
+            variant="outlined"
+          />
+        </Box>
+        <Divider />
+        {loading ? (
+          <Box
+            sx={{
+              p: 2,
+              minHeight: '400px',
+              alignContent: 'center',
+              display: 'flex',
+              justifyContent: 'center'
+            }}
+          >
+            <CircularProgress size={100} />
+          </Box>
+        ) : (
+          <Scrollbar>
+            <Box
+              sx={{
+                p: 2,
+                minHeight: '400px'
+              }}
+            >
+              {editOn ? (
+                <EditTable
+                  fields={fields}
+                  fieldTypeOptions={fieldTypeOptions}
+                  saveChanges={saveChanges}
+                  formUri={metadataForm.uri}
+                />
+              ) : (
+                <DisplayTable fields={fields} startEdit={startEdit} />
+              )}
+            </Box>
+          </Scrollbar>
+        )}
+      </Card>
+    </Box>
+  );
+};
+
+MetadataFormFields.propTypes = {
+  metadataForm: PropTypes.any.isRequired,
+  visibilityDict: PropTypes.any.isRequired,
```

Comment:
nit: do we use `visibilityDict` in `MetadataFormFields` ?

Reply:
[286118c](https://github.com/data-dot-all/dataall/pull/1455/commits/286118c31d1b756b01bd24d823f283f9ea27bd32)